src/FSharpPlus/Data/NonEmptyList.fs

```diff
@@ -215,20 +585,298 @@ module NonEmptyList =
+    /// <param name="list">The input list.</param>
+    /// <returns>The sorted list.</returns>
+    let sortWith (comparer: 'T -> 'T -> int) (list: NonEmptyList<'T>) : NonEmptyList<'T> = 
+        Seq.sortWith comparer list |> ofSeq
+    
+    /// <summary>Splits the list at the specified index.</summary>
+    /// <param name="index">The index at which to split the list.</param>
+    /// <param name="list">The input list.</param>
+    /// <returns>A tuple containing the two lists.</returns>
+    let splitAt (index: int) (list: NonEmptyList<'T>) : NonEmptyList<'T> * NonEmptyList<'T> = 
+        list |> toList |> List.splitAt index |> fun (a, b) -> (ofList a, ofList b)
+    
+    /// <summary>Splits the list into the specified number of lists.</summary>
+    /// <param name="count">The number of lists to create.</param>
+    /// <param name="list">The input list.</param>
+    /// <returns>A list of lists.</returns>
+    let splitInto (count: int) (list: NonEmptyList<'T>) : NonEmptyList<NonEmptyList<'T>> = 
+        Seq.splitInto count list |> Seq.map ofSeq |> ofSeq
+    
+    /// <summary>Computes the sum of the elements of the list.</summary>
+    /// <param name="list">The input list.</param>
+    /// <returns>The sum of the elements.</returns>
+    let inline sum list = 
+        Seq.sum list
+    
+    /// <summary>Computes the sum of the elements of the list, using the given projection.</summary>
+    /// <param name="projection">A function to transform the list elements before summing.</param>
+    /// <param name="list">The input list.</param>
+    /// <returns>The sum of the transformed elements.</returns>
+    let inline sumBy projection list = 
+        Seq.sumBy projection list
+        
+    /// <summary>Returns a list that contains the first N elements of the list.</summary>
+    /// <param name="count">The number of elements to take.</param>
+    /// <param name="list">The input list.</param>
+    /// <returns>The result list.</returns>
+    let take (count: int) (list: NonEmptyList<'T>) : NonEmptyList<'T> = 
+        Seq.take count list |> ofSeq
+    
+    /// <summary>Returns a list that contains the elements of the list while the predicate is true.</summary>
+    /// <param name="predicate">A function to test each element of the list.</param>
+    /// <param name="list">The input list.</param>
+    /// <returns>The result list.</returns>
+    let takeWhile (predicate: 'T -> bool) (list: NonEmptyList<'T>) : NonEmptyList<'T> = 
+        Seq.takeWhile predicate list |> ofSeq
+    
+    /// <summary>Truncates the list to the specified length.</summary>
+    /// <param name="count">The maximum number of elements to include in the list.</param>
+    /// <param name="list">The input list.</param>
+    /// <returns>The truncated list.</returns>
+    let truncate (count: int) (list: NonEmptyList<'T>) : NonEmptyList<'T> = 
+        Seq.truncate count list |> ofSeq
```

Comment:
These may also produce empty lists. For `take` and `truncate`, we could check if `count` is positive.